File: app/src/main/java/com/example/mytemplate/utils/SessionManager.java
package com.example.mytemplate.utils;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;

import androidx.annotation.Nullable;

import com.example.mytemplate.config.GlobalConfig;

public class SessionManager {

    private static final String KEY_TOKEN = "key_token";
    private static final String KEY_NAME = "key_name";
    private static final String KEY_USERNAME = "key_username";

    @SuppressLint("CommitPrefEdits")
    public static void setSession(Context context, Bundle data) {
        if(context == null) return;
        if(data == null) return;

        // set editor
        SharedPreferences.Editor editor = getSessionSharedPreference(context).edit();

        editor.putString(KEY_TOKEN, data.getString(KEY_TOKEN));
        editor.putString(KEY_NAME, data.getString(KEY_NAME));
        editor.putString(KEY_USERNAME, data.getString(KEY_USERNAME));
        editor.apply();
        editor.commit();
    }

    @SuppressLint("CommitPrefEdits")
    public static void clearSession(Context context){
        if(context == null) return;

        // clear shared preference
        getSessionSharedPreference(context).edit().clear().apply();
    }

    @Nullable
    public static String getToken(Context context){
        if(context == null) return null;

        return getSessionSharedPreference(context).getString(KEY_TOKEN, null);
    }

    @Nullable
    public static String getName(Context context){
        if(context == null) return null;

        return getSessionSharedPreference(context).getString(KEY_NAME, null);
    }

    @Nullable
    public static String getUsername(Context context){
        if(context == null) return null;

        return getSessionSharedPreference(context).getString(KEY_USERNAME, null);
    }

    public static boolean isExist(Context context){
        String token = getToken(context);
        return token != null && !token.isEmpty();
    }

    private static SharedPreferences getSessionSharedPreference(Context context){
        return context.getSharedPreferences(GlobalConfig.getSharePreferenceSession(), Context.MODE_PRIVATE);
    }
}

File: app/src/main/java/com/example/mytemplate/utils/SessionManager.kt
package com.example.mytemplate.utils

import android.annotation.SuppressLint
import android.content.Context
import android.content.SharedPreferences
import android.os.Bundle
import com.example.mytemplate.config.GlobalConfig

class SessionManager(private val context: Context) {

    companion object{
        const val KEY_TOKEN = "key_token"
        const val KEY_NAME = "key_name"
        const val KEY_USERNAME = "key_username"
    }

    @SuppressLint("CommitPrefEdits")
    fun setSession(data: Bundle?) {
        data?.let {
            // set editor
            val editor = getSessionSharedPreference().edit()
            editor.putString(KEY_TOKEN, data.getString(KEY_TOKEN))
            editor.putString(KEY_NAME, data.getString(KEY_NAME))
            editor.putString(KEY_USERNAME, data.getString(KEY_USERNAME))
            editor.apply()
            editor.commit()
        }
    }

    /**
     * Clear shared preference
     */
    @SuppressLint("CommitPrefEdits")
    fun clearSession(): Unit = getSessionSharedPreference().edit().clear().apply()

    fun getToken(): String? = getSessionSharedPreference().getString(KEY_TOKEN, null)

    fun getName(): String? = getSessionSharedPreference().getString(KEY_NAME, null)

    fun getUsername(): String? = getSessionSharedPreference().getString(KEY_USERNAME, null)

    fun isExist(): Boolean {
        val token = getToken()
        return token != null && token.isNotEmpty()
    }

    private fun getSessionSharedPreference() = context.getSharedPreferences(GlobalConfig.sharePreferenceSession, Context.MODE_PRIVATE)
}